### PR TITLE
Course enrichment domain

### PR DESCRIPTION
**Pull request by timabell**
### Context

Stub endpoint for course enrichment for coding against

### Changes proposed in this pull request

3 new enrichment endpoints - course enrichment get, set, publish

They don't do anything yet.

### CODE REVIEW

File: src/ManageCourses.Api/Model/UcasCourseEnrichmentGetModel.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace GovUk.Education.ManageCourses.Api.Model
+{
+    public class UcasCourseEnrichmentGetModel
+    {
+        public UcasCourseEnrichmentGetModel()
+        {
+            EnrichmentModel = new CourseEnrichmentModel();
+        }
+        public CourseEnrichmentModel EnrichmentModel { get; set; }
```
Comment (the-daniel-rothig) [inline src/ManageCourses.Api/Model/UcasCourseEnrichmentGetModel.cs]: Doesn't this need identifying informmation aka. a `courseCode` and a `providerCode`?

**Comment by the-daniel-rothig** [comment]
Oh also, you probably want to bump the API version number to `0.6.1`

**Comment by defong** [comment]
No @the-daniel-rothig MINOR version when you add functionality in a backwards-compatible manner

**Comment by timabell** [comment]
@the-daniel-rothig re versions
I beg to differ
> MINOR version when you **add functionality** in a backwards-compatible manner, and
> PATCH version when you make backwards-compatible bug fixes.

**Comment by the-daniel-rothig** [comment]
Oops I stand corrected, sry.

File: src/ManageCourses.Api/Model/UcasCourseEnrichmentGetModel.cs (hunk 1 above)
Comment (timabell) [reply src/ManageCourses.Api/Model/UcasCourseEnrichmentGetModel.cs]: They are in the url, captured as separate string parameters.
Just tried out some variations locally with the fields added to the model.

* Adding to model: now have duplicate places to post this info, bits of url are not bound to fields of model
* Adding to model and removing controller action params: results in broken url pattern (replacement bits disappear, losing our nice restful url setup)

**Pull request merged**